drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/AbstractScenarioSimulationGridPanelHandler.java

```diff
@@ -0,0 +1,154 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.handlers;
+
+import com.ait.lienzo.client.core.types.Point2D;
+import org.drools.workbench.screens.scenariosimulation.client.metadata.ScenarioHeaderMetaData;
+import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGrid;
+import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGridColumn;
+import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGridPanel;
+import org.uberfire.ext.wires.core.grids.client.widget.grid.renderers.grids.impl.BaseGridRendererHelper;
+
+import static org.uberfire.ext.wires.core.grids.client.util.CoordinateUtilities.convertDOMToGridCoordinate;
+import static org.uberfire.ext.wires.core.grids.client.util.CoordinateUtilities.getUiColumnIndex;
+import static org.uberfire.ext.wires.core.grids.client.util.CoordinateUtilities.getUiRowIndex;
+
+public abstract class AbstractScenarioSimulationGridPanelHandler  {
+
+    protected ScenarioGridPanel scenarioGridPanel;
+    protected ScenarioGrid scenarioGrid;
+    protected BaseGridRendererHelper rendererHelper;
+
+    public AbstractScenarioSimulationGridPanelHandler() {
+    }
+
+    public void setScenarioGridPanel(ScenarioGridPanel scenarioGridPanel) {
+        this.scenarioGridPanel = scenarioGridPanel;
+        this.scenarioGrid = scenarioGridPanel.getScenarioGrid();
+        this.rendererHelper = scenarioGrid.getRendererHelper();
+    }
+
+    /**
+     * It calculates the cell related to the given canvas coordinates. These coordinates will be handled by
+     * <code>manageHeaderCoordinates</code> if found cell is an HEADER or <code>manageBodyCoordinates</code> otherwise.
+     * @param canvasX
+     * @param canvasY
+     * @return
+     */
+    protected boolean manageCoordinates(final int canvasX, final int canvasY) {
+        final Point2D gridClickPoint = convertDOMToGridCoordinateLocal(canvasX, canvasY);
+        Integer uiRowIndex = getUiHeaderRowIndexLocal(gridClickPoint);
+        boolean isHeader = true;
+        if (uiRowIndex == null) {
+            uiRowIndex = getUiRowIndexLocal(gridClickPoint.getY());
+            isHeader = false;
+        }
+        final Integer uiColumnIndex = getUiColumnIndexLocal(gridClickPoint.getX());
+        ScenarioGridColumn scenarioGridColumn = uiColumnIndex != null ? (ScenarioGridColumn) scenarioGrid.getModel().getColumns().get(uiColumnIndex) : null;
+        if (isHeader) {
+            return manageHeaderCoordinates(uiColumnIndex, scenarioGridColumn, gridClickPoint);
+        } else {
+            return (uiRowIndex == null || uiColumnIndex == null) ? manageBodyCoordinates(-1, -1) : manageBodyCoordinates(uiRowIndex, uiColumnIndex);
```

Comment:
I would change this line to simple `return manageBodyCoordinates(uiRowIndex, uiColumnIndex)` and add the check for null values inside of `manageBodyCoordinates`. This wil facilitate invocation from other places (if needed in future), what do you think?

Reply:
Agree, DONE!